tests/test_minimal/test_tools/test_backend_and_dataset_configuration/test_zarr_dataset_configuration_model.py

```diff
@@ -0,0 +1,280 @@
+    with pytest.raises(ValueError) as error_info:
+        ZarrDatasetConfiguration(
+            dataset_info=mock_DatasetInfo(),
+            chunk_shape=(78_126, 64),
+            buffer_shape=(78_125, 384),
+        )
+
+    expected_error = (
+        "Some dimensions of the chunk_shape=(78126, 64) exceed the buffer_shape=(78125, 384))! (type=value_error)"
+    )
+    assert expected_error in str(error_info.value)
+
+
+def test_validator_buffer_shape_exceeds_full_shape():
+    with pytest.raises(ValueError) as error_info:
+        ZarrDatasetConfiguration(
+            dataset_info=mock_DatasetInfo(),
+            chunk_shape=(78_125, 64),
+            buffer_shape=(1_250_000, 385),
+        )
+
+    expected_error = (
+        "Some dimensions of the buffer_shape=(1250000, 385) exceed the full_shape=(1800000, 384)! (type=value_error)"
+    )
+    assert expected_error in str(error_info.value)
+
+
+def test_validator_chunk_dimensions_do_not_evenly_divide_buffer():
+    with pytest.raises(ValueError) as error_info:
+        ZarrDatasetConfiguration(
+            dataset_info=mock_DatasetInfo(),
+            chunk_shape=(78_125, 7),
+            buffer_shape=(1_250_000, 384),
+        )
+
+    expected_error = (
+        "Some dimensions of the chunk_shape=(78125, 7) do not evenly divide the buffer_shape=(1250000, 384))! "
+        "(type=value_error)"
+    )
+    assert expected_error in str(error_info.value)
+
+
+def test_validator_filter_options_has_methods():
+    with pytest.raises(ValueError) as error_info:
+        ZarrDatasetConfiguration(
+            dataset_info=mock_DatasetInfo(),
+            chunk_shape=(78_125, 64),
+            buffer_shape=(1_250_000, 384),
+            filter_methods=None,
+            filter_options=[dict(clevel=5)],
+        )
+
+    expected_error = (
+        "`filter_methods` is `None` but `filter_options` is not (received [{'clevel': 5}])! (type=value_error)"
+    )
+    assert expected_error in str(error_info.value)
+
+
+def test_validator_filter_methods_length_match_options():
+    with pytest.raises(ValueError) as error_info:
+        ZarrDatasetConfiguration(
+            dataset_info=mock_DatasetInfo(),
+            chunk_shape=(78_125, 64),
+            buffer_shape=(1_250_000, 384),
+            filter_methods=["blosc", "delta"],
+            filter_options=[dict(clevel=5)],  # Correction would be to add a second element `dict()` to avoid ambiguity
+        )
+
+    expected_error = (
+        "Length mismatch between `filter_methods` (2 methods specified) and `filter_options` (1 options found)! "
+        "These two must match one-to-one. (type=value_error)"
+    )
+    assert expected_error in str(error_info.value)
+
+
+def test_available_hdf5_compression_methods_not_empty():
+    assert len(AVAILABLE_ZARR_COMPRESSION_METHODS) > 0
+
+
+def test_default_compression_is_always_available():
+    assert "gzip" in AVAILABLE_ZARR_COMPRESSION_METHODS
+
+
+def test_mutation_validation():
```

Comment:
That was my thining too - not exhaustive testing since I assume Pydantic handles that - but for my own sanity I want to make sure it is enabled properly by us

Reply:
Yes. I this line of reasoning makes sense to me.